File: java/short-url/src/main/java/com/xg/shorturl/dao/ShortUrlDao.java
package com.xg.shorturl.dao;

/**
 * 短链接数据访问层
 * @author xionggen
 */
public interface ShortUrlDao {

    /**
     * 根据短链接查询原始连接
     * @param shortUrl 短连接
     * @return 原始连接
     */
    String queryOriginalUrl(String shortUrl);

    /**
     * 根据原始链接查询短链接
     * @param originalUrl 原始链接
     * @return 短链接
     */
    String queryShortUrl(String originalUrl);

    /**
     * 保存短链接与原始链接
     * @param shortUrl 短链接
     * @param originalUrl 原始链接
     * @return 1-成功，0-失败
     */
    int save(String shortUrl, String originalUrl);
}

File: fullstack/ts-short-url/src/dao/shorturldao.ts

/**
 * / route
 *
 * @class ShortUrlDao
 */
export class ShortUrlDao  {

    private originalUrlyMap: Map<string, string>;   // key(originalUrl),value(shortUrlID)
    private shortUrlIDMap: Map<string, string>;   // key(shortUrlID),value(originalUrl)

    constructor() {
        this.originalUrlyMap = new Map<string, string>();
        this.shortUrlIDMap = new Map<string, string>();
    }
    
    public queryOriginalUrl(shortUrl: string): string {
        if (this.shortUrlIDMap.has(shortUrl)) {
            return this.shortUrlIDMap.get(shortUrl) as string;
        }

        return "";
    }

    public queryShortUrl(originalUrl: string): string {
        if (this.originalUrlyMap.has(originalUrl)) {
            return this.originalUrlyMap.get(originalUrl) as string;
        }
        return "";
    }

    public save(shortUrl: string, originalUrl: string): boolean {

        this.originalUrlyMap.set(originalUrl, shortUrl);
        this.shortUrlIDMap.set(shortUrl, originalUrl);
        return true;
    }    
}